docs/flow/modin/backends/index.rst

```diff
@@ -0,0 +1,46 @@
+:orphan:
+
+Query compiler
+==============
+
+Modin has several execution backends. Calling any DataFrame API function will end up in
+some backend-specific method. Query compiler is a bridge between Modin DataFrame and
+an actual execution engine.
+
+.. image:: /img/simplified_query_flow.svg
+    :align: right
+    :width: 300px
+
+Query compilers of all backends provide a common API, which is used by Modin DataFrame
+to emulate pandas API. The role of the query compiler is to translate its API into
```

Comment:
```suggestion
to support dataframe queries. The role of the query compiler is to translate its API into
```

Reply:
Does DataFrame uses a query compiler to support "dataframe queries"? What are "dataframe queries" and why we should use this term instead of "pandas API"?

Reply:
The Modin implementation is more general than just pandas. Since this is developer documentation, we can assume a higher technical level and talk more generally.